sys/net/transport_layer/ng_udp/ng_udp.c

```diff
@@ -72,7 +72,7 @@ static uint16_t _calc_csum(ng_pktsnip_t *hdr, ng_pktsnip_t *pseudo_hdr,
         payload = payload->next;
     }
     /* process applicable UDP header bytes */
-    csum = ng_inet_csum(csum, (uint8_t *)hdr->data, 6);
+    csum = ng_inet_csum(csum, (uint8_t *)hdr->data, sizeof(ng_udp_hdr_t));
```

Comment:
Why? This is wasting cycles for nothing *and* making it less flexible. Now for reusing a received snip you'd have to exlicitly zero the checksum field...